Sets the default architecture of sh4-linux-gdb (when no executable
has been given) to "sh4" rather than "sh". This allows an sh4tp
connection without an explicit "set architecture sh4". The latter
causes problems on st40-300.

Signed-off-by: Chris Smith <chris.smith@st.com>

Index: gdb-6.8/bfd/cpu-sh.c
===================================================================
--- gdb-6.8.orig/bfd/cpu-sh.c
+++ gdb-6.8/bfd/cpu-sh.c
@@ -337,14 +337,14 @@ const bfd_arch_info_type bfd_sh_arch =
   32,				/* 32 bits in an address.  */
   8,				/* 8 bits in a byte.  */
   bfd_arch_sh,
-  bfd_mach_sh,
+  bfd_mach_sh4,
   "sh",				/* Architecture name.   */
-  "sh",				/* Machine name.  */
+  "sh4",				/* Machine name.  */
   1,
   TRUE,				/* The default machine.  */
   bfd_default_compatible,
   bfd_default_scan,
-  SH_NEXT
+  SH4_NEXT
 };
 
 
